plugins/modules/aci_interface_policy_pim.py

```diff
@@ -0,0 +1,445 @@
+    authentication_key = module.params.get("authentication_key")
+    authentication_type = MATCH_PIM_INTERFACE_POLICY_AUTHENTICATION_TYPE_MAPPING.get(module.params.get("authentication_type"))
+    description = module.params.get("description")
+    name_alias = module.params.get("name_alias")
+    state = module.params.get("state")
+
+    designed_router_delay = module.params.get("designed_router_delay")
+    if designed_router_delay is not None and designed_router_delay not in range(1, 65536):
+        module.fail_json(msg="Parameter 'designed_router_delay' is only valid in range between 1 and 65535.")
+
+    designed_router_priority = module.params.get("designed_router_priority")
+    if designed_router_priority is not None and designed_router_priority not in range(1, 4294967296):
+        module.fail_json(msg="Parameter 'designed_router_priority' is only valid in range between 1 and 4294967295.")
+
+    hello_interval = module.params.get("hello_interval")
+    if hello_interval is not None and hello_interval not in range(1, 18724287):
+        module.fail_json(msg="Parameter 'hello_interval' is only valid in range between 1 and 18724286.")
+
+    join_prune_interval = module.params.get("join_prune_interval")
+    if join_prune_interval is not None and join_prune_interval not in range(60, 65521):
+        module.fail_json(msg="Parameter 'join_prune_interval' is only valid in range between 60 and 65520.")
+
+    if module.params.get("control_state"):
+        control_state = ",".join([MATCH_PIM_INTERFACE_POLICY_CONTROL_STATE_MAPPING.get(v) for v in module.params.get("control_state")])
+    else:
+        control_state = None
+
+    child_classes = dict(
+        pimJPInbFilterPol=module.params.get("inbound_join_prune_filter_policy"),
+        pimJPOutbFilterPol=module.params.get("outbound_join_prune_filter_policy"),
+        pimNbrFilterPol=module.params.get("neighbor_filter_policy"),
+    )
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="pimIfPol",
+            aci_rn="pimifpol-{0}".format(pim),
+            module_object=pim,
+            target_filter={"name": pim},
+        ),
+        child_classes=list(child_classes.keys()),
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = []
+        for class_name, class_input in child_classes.items():
+            if class_input is not None:
+                if class_input == "" and isinstance(aci.existing, list) and len(aci.existing) > 0:
+                    for child in aci.existing[0].get("pimIfPol", {}).get("children", {}):
+                        if child.get(class_name):
+                            child_configs.append(
+                                {
+                                    class_name: dict(
```

Comment:
Shouldn't this be?
class_name=dict(

Reply:
no you can't cause class_name is a variable here. It would not work with the previous way to state dict (like you mentioned) unless I missed something.

Reply:
In that case, for consistency I might consider using the below:

```
child_configs.append(
    dict([(class_name, dict(attributes=dict(status="deleted")))])
)
```

Reply:
ok ok thank you, will do that!